### PR TITLE
chore(release): 2.0.0

### PR DESCRIPTION
**Pull request by pallabmaiti**
## Description of the change

1. replaced `v1/track`, `v1/page`, `v1/alias`, `v1/screen`, `v1/identify`, `v1/track` by `v1/batch`
2. gzip support
3. `screen` API
4. max batch size of 500 KB
5. max message size of 32 KB

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,5 +2,6 @@
 
 source 'https://rubygems.org'
 
-# Specify your gem's dependencies in simple_segment.gemspec
+# Specify your gem's dependencies in rudder_analytics_sync.gemspec
+gem "debug"
```
Comment (arnab-p) [inline Gemfile]: We probably don't want to keep it as a depencency

**Review by arnab-p — APPROVED** [review]

File: Gemfile (hunk 1 above)
Comment (pallabmaiti) [reply Gemfile]: This is okay. We haven't specified it under the `gemspec` file.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
```
Comment (bardisg) [inline .github/dependabot.yml:3]: lets also add a check for the ruby dependencies too

File: .github/workflows/build-and-quality-checks.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
+          bundler-cache: true
+      - name: Install dependencies
+        run: bundle install
+      - name: Run tests
+        run: bundle exec rake
+      # - name: Install rubocop
```
Comment (bardisg) [inline .github/workflows/build-and-quality-checks.yml:22]: why we do not run this one too?

File: .github/workflows/check-pr-title.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Check PR title
+on:
+  pull_request:
+    branches: ['v3.3']
```
Comment (bardisg) [inline .github/workflows/check-pr-title.yml]: add master

File: .github/workflows/slack-notify.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  deploy-tag:
+    name: Notify Slack
+    runs-on: ubuntu-latest
+    steps:
+      - name: Send message to Slack channel
+        id: slack
+        uses: slackapi/slack-github-action@v1.23.0
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+          PROJECT_NAME: 'Go SDK'
```
Comment (bardisg) [inline .github/workflows/slack-notify.yml]: correct name

File: .github/workflows/build-and-quality-checks.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+name: Code Quality Checks
```
Comment (bardisg) [inline .github/workflows/build-and-quality-checks.yml:1]: please also add the housekeeping action. copy paste from js sdk repo and change branch names

File: .github/dependabot.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+version: 2
```
Comment (bardisg) [inline .github/dependabot.yml:1]: can we add a codeowners file too?

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,16 +6,36 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+## [1.5.0] - 2022-12-06
+
+### Changed
+- Use the /v1/batch endpoint for batching events (this should be backward compatible according to Segment) https://github.com/whatthewhat/simple_segment/pull/40 by [@sgallag-insta](https://github.com/sgallag-insta)
```
Comment (bardisg) [inline CHANGELOG.md]: please remove the comment for segment

File: .github/workflows/build-and-quality-checks.yml (hunk 3 above)
Comment (pallabmaiti) [reply .github/workflows/build-and-quality-checks.yml:22]: This is not working.

File: .github/workflows/build-and-quality-checks.yml (hunk 6 above)
Comment (pallabmaiti) [reply .github/workflows/build-and-quality-checks.yml:1]: This SDK is not actively developed. I am not in favor of the housekeeping actions for this repo.

File: .github/dependabot.yml (hunk 7 above)
Comment (pallabmaiti) [reply .github/dependabot.yml:1]: We can. Who will be the CODEOWNERs?

File: CHANGELOG.md (hunk 8 above)
Comment (pallabmaiti) [reply CHANGELOG.md:12]: This is from the forked. Should we remove them?

Comment (pallabmaiti) [reply CHANGELOG.md]: Let's remove the CHANGELOG. It doest match our releases.

File: .github/dependabot.yml (hunk 2 above)
Comment (pallabmaiti) [reply .github/dependabot.yml:3]: Can you help me with this?

Comment (bardisg) [reply .github/dependabot.yml:3]: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem list all supported package-ecosystem add another section for the ruby one

**Comment by clarkedb** [comment]
@pallabmaiti This repo does not support issues or discussions, so I wasn't sure where to post about this, but I think you should know that the v2.0.0 build is broken. The changes made in [this commit](https://github.com/rudderlabs/rudder-sdk-ruby-sync/pull/6/commits/da3c5b8ff3d9a21a022eaf5d8850f892fe00e0e7) broke all operations.

For example, now calling `track` will lead to an `ArgumentError` because `Track#call` calls `Request#post` like so

```ruby
request.post('/v1/track', batch: [batch])
```

but `Request#post` does not take a parameter `:batch`. It takes two positional arguments and one keyword argument `:headers`.

Upgrading to v2.0.0 will lead to `ArgumentError: wrong number of arguments (given 1, expected 2)` being raised from within the Rudderstack ruby SDK gem.

**Comment by pallabmaiti** [comment]
@clarkedb I opened the Issue tab. Can you post this there, and can you send what are you trying to send?

**Comment by clarkedb** [comment]
> @clarkedb I opened the Issue tab. Can you post this there, and can you send what are you trying to send?

@pallabmaiti I've opened issue #9 per your request

**Pull request merged**